app/models/post.rb

```diff
@@ -1,5 +1,25 @@
 class Post < ActiveRecord::Base
   belongs_to :author, class_name: "User"
+  has_many :taggings
+  has_many :tags, through: :taggings
 
+  validate :all_tags_are_valid
   validates :author, :body, :title, presence: true
+
+  def all_tags=(names)
+    names.split(",").map do |name|
+      self.tags << Tag.where(name: name.strip).first_or_create
```

Comment:
This should be an `=` not a `<<`.  The current implementation prevents ever removing tag (for example once edit is implemented).